scipy/sparse/linalg/_eigen/tests/test_svds.py

```diff
@@ -588,6 +633,37 @@ def reorder(args):
                                     np.dot(U2, np.dot(np.diag(s2), VH2)),
                                     rtol=eps)
 
+    SHAPES = ((100, 100), (100, 101), (101, 100))
+
+    @pytest.mark.filterwarnings("ignore:Exited at iteration")
+    @pytest.mark.parametrize("shape", SHAPES)
+    # ARPACK supports only dtype float, complex, or np.float32
+    @pytest.mark.parametrize("dtype", (float, complex, np.float32))
+    def test_small_sigma_sparse(self, shape, dtype):
```

Comment:
So basically there is one solver that can handle this in a general way. Are we really sure that we want this refinement in sparse library? This is not even really sparse and it will break many existing use of `svds` in the name of saving some small svds.

Reply:
@ilayn could you please elaborate with constructive suggestions, like action items for me to improve/delete in this test?